### PR TITLE
Added regression tests

### LINKED ISSUE #117: Add regression tests for different types of contract upgrades
As suggested by Nomic (https://github.com/zeppelinos/zos-lib/pull/102) we should have regression tests to ensure different types of contract upgrades work fine, such as:
- can add/remove a function,
- can change a function signature,
- can add/remove fallback function.

They seem obvious given the current implementation but it would allow us to make changes to the `Proxy` family of contracts with confidence.

### PR DESCRIPTION
**Pull request by filiplazovic**
Closes #117

Adding/removing functions
Changing function signature
Adding/removing fallback function

### CODE REVIEW

**Comment by filiplazovic** [comment]
Since no one cares about this, im closing it

**Comment by facuspagnuolo** [comment]
@filiplazovic sorry for the big delay, I'm reopening this issue to review it asap. Thanks for the contribution!

File: contracts/mocks/Implementation4.sol [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  function getValue() public view returns (uint) {
+    return value;
+  }
+
+  function() public {
+    value = 1;
+  }
+}
```
Comment (facuspagnuolo) [inline contracts/mocks/Implementation4.sol]: Would you mind placing all the `ImplementationN` mocks in one single file?

File: test/contracts/regression/AdminUpgradeabilityProxy.test.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    await assertRevert(sendTransaction(proxy, '', [], [], {
+      from: anotherAddress
+    }));
+
+    const proxyInstance2 = Implementation2.at(proxy.address);
+
+    const res = await proxyInstance2.getValue.call();
+    assert.equal(res.toString(), "0");
+  });
+});
```
Comment (facuspagnuolo) [inline test/contracts/regression/AdminUpgradeabilityProxy.test.js]: Nit, but I would move this tests into a `'regression'` context within the `contracts/upgradeability/AdminUpgradeabilityProxy.test.js` file

File: test/contracts/regression/AdminUpgradeabilityProxy.test.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    const proxyInstance1 = await Implementation1.at(proxy.address);
+    await proxyInstance1.setValue(42);
+
+    const instance2 = await Implementation2.new();
+    await proxy.upgradeTo(instance2.address, {
+      from: admin
+    });
+
+    const proxyInstance2 = Implementation2.at(proxy.address);
+    const res = await proxyInstance2.getValue.call();
```
Comment (facuspagnuolo) [inline test/contracts/regression/AdminUpgradeabilityProxy.test.js]: you can use `await proxyInstance2.getValue()` instead

File: test/contracts/regression/AdminUpgradeabilityProxy.test.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    const proxyInstance1 = await Implementation1.at(proxy.address);
+    await proxyInstance1.setValue(42);
+
+    const instance3 = await Implementation3.new();
+    await proxy.upgradeTo(instance3.address, {
+      from: admin
+    });
+
+    const proxyInstance3 = Implementation3.at(proxy.address);
+    const res = await proxyInstance3.getValue.call(8);
```
Comment (facuspagnuolo) [inline test/contracts/regression/AdminUpgradeabilityProxy.test.js]: and here you can use `await proxyInstance3.getValue(8)`

**Comment by filiplazovic** [comment]
@facuspagnuolo Updated! Thanks for the feedback

**Pull request merged**